src/pixelator/graph/backends/implementations.py

```diff
@@ -428,9 +517,118 @@ def __init__(
         """
         self._raw = raw
 
+    @staticmethod
+    def _build_plain_graph_from_edgelist(
+        df: pl.LazyFrame,
+        create_using: Union[nx.Graph, nx.MultiGraph],
+    ) -> Union[nx.Graph, nx.MultiGraph]:
+        g = nx.empty_graph(0, create_using)
+        g.add_edges_from(
+            (row[0], row[1])
+            for row in df.collect(streaming=True).iter_rows(
+                named=False, buffer_size=1000
+            )
+        )
+        return g
+
+    @staticmethod
+    def _build_graph_with_node_counts_from_edgelist(
+        df: pl.LazyFrame,
+        create_using: bool,
+    ) -> Union[nx.Graph, nx.MultiGraph]:
+        unique_markers = set(df.unique("marker").collect()["marker"].to_list())
+        initial_marker_dict = {marker: 0 for marker in unique_markers}
+
+        g: nx.Graph = nx.empty_graph(0, create_using)
+
+        for row in (
+            df.select(["upia", "upib", "marker"])
+            .collect(streaming=True)
+            .iter_rows(named=False, buffer_size=1000)
+        ):
+            # We are duplicating code here, since it gives
+            # a performance boost of roughly 10% here.
+            # Which isn't that much, but will add up when
+            # we have large edge lists.
+            node_1 = row[0]
+            node_2 = row[1]
+            marker = row[2]
+
+            existing_node_1 = g.nodes.get(node_1)
+            if existing_node_1:
+                existing_node_1["markers"][marker] += 1
+            else:
+                marker_dict = initial_marker_dict.copy()
+                marker_dict[marker] += 1
+                g.add_node(node_1, markers=marker_dict)
+
+            existing_node_2 = g.nodes.get(node_2)
+            if existing_node_2:
+                existing_node_2["markers"][marker] += 1
+            else:
+                marker_dict = initial_marker_dict.copy()
+                marker_dict[marker] += 1
+                g.add_node(node_2, markers=marker_dict)
+
+            g.add_edge(node_1, node_2)
+
+        return g
+
+    @staticmethod
+    def _add_node_attributes(
+        graph: Union[nx.Graph, nx.MultiGraph], a_nodes: set[str]
+    ) -> None:
+        node_names = {node: node for node in graph.nodes()}
+        pixel_type = {node: "A" if node in a_nodes else "B" for node in graph.nodes()}
+        type_ = {node: node in a_nodes for node in graph.nodes()}
```

Comment:
`type_`is only setting A-nodes? Back-underscoring variables is interesting, standard?

Reply:
I do this because `type` is a reserved keyword in python, so then the standard of to add a `_`.